ui/src/app/store/subnet/utils.ts

```diff
@@ -63,3 +63,39 @@ export const getSubnetsInSpace = (
   subnets: Subnet[],
   spaceId: Space["id"]
 ): Subnet[] => subnets.filter((subnet) => subnet.space === spaceId);
+
+/**
+ * Get VLAN on a given subnet
+ * @param subnets - Subnets.
+ * @param vlanId - VLAN id.
+ * @return Subnets for a given VLAN id.
+ */
+export const getVLANOnSubnet = (
+  vlans: VLAN[],
+  subnet: Subnet
+): VLAN | undefined => vlans.find((vlan) => vlan.id === subnet.vlan);
+
+/**
+ * Get if DHCP is enabled on a given subnet
+ * @param subnet - All VLANS.
+ * @param subnet - The subnet to check.
+ */
+export const getIsDHCPEnabled = (vlans?: VLAN[], subnet?: Subnet): boolean => {
+  if (!vlans || !subnet) {
+    return false;
+  }
+  const vlanOnSubnet = getVLANOnSubnet(vlans, subnet);
+  return vlanOnSubnet?.dhcp_on || false;
+};
+
+/**
+ * Get if a subnet can be deleted.
+ * @param subnet - The subnet to check.
+ */
+export const getCanBeDeleted = (vlans: VLAN[], subnet: Subnet): boolean => {
```

Comment:
I'd consider making this and `getIsDHCPEnabled ` be hooks. That way you could move everything you need for the hook here e.g. dispatching to fetch the vlans, selecting the vlans from the store. That way you could also make `getVLANOnSubnet` a selector (we already have the inverse as a selector: https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/store/subnet/selectors.ts#L106).

Reply:
I agree, but wouldn't hurt to have an underlying util function anyway to handle additional use cases?

Reply:
Added a task for this: https://github.com/canonical-web-and-design/app-tribe/issues/720